R/get-pkgs.R

```diff
@@ -5,20 +5,16 @@ deepstate_datatype_check <- function(params.list){
   params.list <- gsub("const","",params.list)
   params.list <-gsub("Rcpp::","",params.list)
   params.list <-gsub(" ","",params.list)
-datatypes <- list("NumericVector","NumericMatrix" ,"arma::mat","double",
+  datatypes <- list("NumericVector","NumericMatrix" ,"arma::mat","double",
                   "string","CharacterVector","int","IntegerVector")
-for(pkg.i in seq_along(params.list)){
-  #print(params.list[[pkg.i]])
-  if(is.element(params.list[[pkg.i]], datatypes) =="FALSE"){
-    return(0);
+  matched <- params.list %in% datatypes
+  if (!all(matched)){
+    mismatched_datatypes <- params.list[!matched]
+    return(list(0,mismatched_datatypes))
   }
-  
-}
-return(1);
-}
-
-
 
+  return(list(1,c()));
```

Comment:
what is this?

Reply:
explicit return is not needed at end of function, please remove.

Reply:
I completely reviewed the entire function and changed it's name to `deepstate_get_mismatched_datatypes`. Now his function returns the list of mismatched arguments, instead of returning 0 or 1 alongside with the mismatched datatypes.